cmd/zed/agents/zfs_mod.c

```diff
@@ -619,6 +687,71 @@ devphys_iter(const char *physical, const char *devid, zfs_process_func_t func,
 	return (data.dd_found);
 }
 
+/*
+ * Given a device identifier, find any vdevs with a matching by-vdev
+ * path.  Normally we shouldn't need this as the comparison would be
+ * made earlier in the devphys_iter().  For example, if we were replacing
+ * /dev/disk/by-vdev/L28, normally devphys_iter() would match the
+ * ZPOOL_CONFIG_PHYS_PATH of "L28" from the old disk config to "L28"
+ * of the new disk config.  However, we've seen cases where
+ * ZPOOL_CONFIG_PHYS_PATH was not in the config for the old disk.  Here's
```

Comment:
As I recall the `devid` and `phys_path` are updated by `fix_paths` during an import, and probably also when doing a replace or add.  If these keys are missing, that it seems like these paths couldn't be determined then.  Or is it also possible that they get cleaned when force faulting a device?

Reply:
I not sure why they were cleared in my particular case.  At the time, I was testing autoreplace on a force faulted disk using various replacement disk configurations (partitioned, labeled, etc).  It's possible that one of those failed replacements (like attempting to replace with a partitioned disk) somehow re-wrote the vdev label to not have a PHYS_PATH or ENC_SYSFS_PATH.  That's just a theory though, I have no proof.